app/controllers/admin/user_creations_controller.rb

```diff
@@ -37,8 +37,11 @@ def set_spam
     action = "mark as " + (params[:spam] == "true" ? "spam" : "not spam")
     AdminActivity.log_action(current_admin, @creation, action: action, summary: @creation.inspect)    
     if params[:spam] == "true"
+      unless @creation.hidden_by_admin
+        @creation.notify_of_hiding_for_spam if @creation_class == Work
+        @creation.hidden_by_admin = true
+      end
       @creation.mark_as_spam!
-      @creation.update_attribute(:hidden_by_admin, true)
```

Comment:
This change is suspicious to me. update_attribute does the change and saves it. The `hidden_by_admin = true` makes the change, but doesn't save it and #save isn't called anywhere later.

This might explain the test that was failing

Reply:
I think the `update_attribute(:spam, true)` inside the `@creation.mark_as_spam!` function saves it. This was a deliberate change because otherwise it would have been saved twice, so the after_save hook would have been called twice, and two emails would have been sent, the hidden-for-spam email first then the tos-violation email second.

The test was failing because I changed the test steps and it caused an unrelated test elsewhere to fail

Reply:
The `Given /^the spam work "([^\"]*)"$/` step? I am confused why that needed modification, the `before_save :hide_spam` should set `hidden_by_admin` even with your changes

Reply:
The `Given /^the spam work "([^\"]*)"$/` step needed to be changed because I wanted to test cases where spam works aren't hidden automatically

> the before_save :hide_spam should set hidden_by_admin even with your changes

Only if the `admin_settings.hide_spam?` is set to true. The 'mark as spam' button is supposed to hide the work regardless of the admin setting.

Reply:
Oh sorry I just realized I misunderstood you there, yeah you're right, I'll try rolling back the test step definition changes when I get a chance